### DISCUSSION TITLE
Add support for functions in script files?

### DISCUSSION THREAD

**Discussion by seanmcleod** [JSBSim-Team/jsbsim#1108] [Ideas]
In discussion - https://github.com/JSBSim-Team/jsbsim/discussions/1099 a user was asking about a mechanism to vary the wind with altitude from a script file. I suggested adding a function in the FDM to vary the wind. However this isn't ideal, i.e. you should be able to vary the wind without having to edit every aircraft's FDM that you want to use.

If the user was running JSBSim via python or some other host environment then they could implement the logic to vary the wind from the host environment.

So I was wondering whether it would be useful to add the capability of being able to define functions within JSBSim script files, e.g. 

```xml
	<run start="0" end="100" dt="0.008333">

		<property value="0"> simulation/notify-time-trigger </property>

		<function name="aero/function/windcalc" copyto="atmosphere/wind-mag-fps">
			<description>Varying wind with altitude</description>
			<table>
				<independentVar>position/h-sl-ft</independentVar>
				<tableData>
					0        0
					40000    40
				</tableData>
			</table>
		</function>

		<event name="Set engines running">
			<condition> simulation/sim-time-sec le 0.1 </condition>
			<set name="propulsion/engine[0]/set-running" value="1"/>
			<set name="propulsion/engine[1]/set-running" value="1"/>
			<set name="propulsion/fuel_freeze" value="1"/>
			<notify/>
		</event>
```

**Comment by gallonmate** [comment]
Seems useful. I would have assumed something like this is already possible to vary the wind via script file.

**Comment by seanmcleod** [comment]
Pretty sure the only way to modify properties in JSBSim XML based scripts is via the `set` element which has a limited set of options compared to a full blown function.

```xml
<set name="propulsion/fuel_freeze" value="1"/>
<set name="fcs/throttle-cmd-norm[0]" action="FG_RAMP" value="0.6" tc="1.0"/>
<set name="fcs/steer-cmd-norm" action="FG_EXP" value="0.0" tc="1"/>
```

```c++
  enum eAction {
    FG_RAMP  = 1,
    FG_STEP  = 2,
    FG_EXP   = 3
  };

  enum eType {
    FG_VALUE = 1,
    FG_DELTA = 2,
    FG_BOOL  = 3
  };
```

**Comment by bcoconni** [comment]
It makes sense and it should not be too difficult to implement.
